src/main/java/com/cloudbees/plugins/credentials/CredentialsResolver.java

```diff
@@ -126,14 +128,14 @@ protected CredentialsResolver(@NonNull Class<F> fromClass) {
             // if the reflective instantiation proves a hot point, put a cache in front.
             try {
                 @SuppressWarnings("unchecked")
-                final CredentialsResolver<Credentials, C> resolver = resolveWith.value().newInstance();
+                final CredentialsResolver<Credentials, C> resolver = resolveWith.value().getDeclaredConstructor().newInstance();
```

Comment:
⁉️ Same here, not sure to understand the added value. More verbose + more exceptions.

Reply:
Resolving here, change reverted to `newInstance()`.